src/main/java/org/luksza/gerrit/rest/PostRetrigger.java

```diff
@@ -65,11 +72,20 @@
   public Object apply(ChangeResource resource, Input input)
       throws AuthException, BadRequestException, ResourceConflictException,
       Exception {
-    HttpClient client = new DefaultHttpClient();
+    CloseableHttpClient client = HttpClients.custom().build();
     Project.NameKey project = resource.getChange().getProject();
-    Optional<String> sessioIdOpt = createSession(client, project);
-    retriggerBuild(input, client, project, sessioIdOpt);
-    return new Output(config.getJenkinsUrl(project));
+    String jenkinsUrl = config.getJenkinsUrl(project);
+    String errorMessage = null;
+    try {
+      Optional<Header> authOpt = getAuth(jenkinsUrl);
+      Optional<Header> crumbOpt = getCrumb(client, project, authOpt);
+      Optional<String> sessionIdOpt = createSession(client, project, crumbOpt, authOpt);
+      retriggerBuild(input, client, project, sessionIdOpt, crumbOpt, authOpt);
+    } catch(IOException e) {
+      errorMessage = e.getMessage();
+    }
+    client.close();
```

Comment:
is there any reason why not to use try-with-resources here? then you don't need to bother with closing this connection at all